core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -344,10 +343,15 @@ private boolean checkForWorkflowCompletion(final Workflow workflow) throws Termi
 
     private List<Task> getNextTask(Workflow workflow, Task task) {
         final WorkflowDef workflowDef = workflow.getWorkflowDefinition();
+        
+        if(TaskType.GOTO.name().equals(task.getTaskType()))
+        	{
+        	    return Collections.emptyList();
+        	}
 
         // Get the following task after the last completed task
         if (SystemTaskType.is(task.getTaskType()) && SystemTaskType.DECISION.name().equals(task.getTaskType())) {
-            if (task.getInputData().get("hasChildren") != null) {
+            if (task.getInputData().get("hasChildren") != null || task.getInputData().get("gotoActivated") != null) {
```

Comment:
What is the significance of gotoActivated?

Reply:
If 'GOTO' task is part of a decision task , then it will take the control of the workflow to the next task mentioned in 'gotoTask' field. The deciderService basically schedule next task defined in DSL, whenever a task (decision task in this case) is completed. Since control is already taken elsewhere by GOTO task, we don't want task subsequent to decision task to run. This property is being added in DecisionTaskMapper, whenever there is a GOTO task in selected branch of decision.